src/anyp/PortCfg.cc

```diff
@@ -101,3 +105,185 @@ AnyP::PortCfg::detailCodeContext(std::ostream &os) const
+{
+    for (const auto &p: PortKindStrings)
+        if (p.first == flags.portKind)
+            return p.second;
+    assert(false); // unreachable
+    return nullptr;
+}
+
+void
+AnyP::PortCfg::rejectFlags(const AnyP::TrafficModeFlags::List &list, const char *detail)
+{
+    const auto &rawFlags = flags.rawConfig();
+    for (const auto &p: list) {
+        if (rawFlags.*p) {
+            debugs(3, DBG_CRITICAL, "FATAL: " << PortOptionStr(p) << " is unsupported on " << PortKindStr(rawFlags) <<  ' ' << detail);
+            self_destruct();
+        }
+    }
+}
+
+void
+AnyP::PortCfg::allowEither(const AnyP::TrafficModeFlags::List &list, const char *detail)
+{
+    const auto &rawFlags = flags.rawConfig();
+    int conflictedCount = 0;
+    SBuf conflicted;
+    for (const auto &p: list) {
+        if (rawFlags.*p) {
+            conflictedCount++;
+            if (!conflicted.isEmpty())
+                conflicted.append(',');
+            conflicted.append(PortOptionStr(p));
+        }
+    }
+    if (conflictedCount <= 1)
+        return;
+
+    debugs(3, DBG_CRITICAL, "FATAL: the combination of " << conflicted <<
+            " is unsupported on " << PortKindStr(rawFlags) << ' ' << detail);
+    self_destruct();
+}
+
+void
+AnyP::PortCfg::checkImplication(const AnyP::TrafficModeFlags::Pointer aFlag, const AnyP::TrafficModeFlags::List &list, const char *detail)
+{
+    const auto &rawFlags = flags.rawConfig();
+    if (!(rawFlags.*aFlag))
+        return;
+
+    for (const auto &p: list) {
+        if (rawFlags.*p)
+            return;
+    }
+
+    debugs(3, DBG_CRITICAL, "FATAL: " << PortOptionStr(aFlag) << " requires one of " << PortOptionStrList(list) <<
+            " on " << PortKindStr(rawFlags) << ' ' << detail);
+    self_destruct();
+}
+
+void
+AnyP::PortCfg::checkImplication(const AnyP::TrafficModeFlags::Pointer aFlag, const AnyP::TrafficModeFlags::Pointer otherFlag, const char *detail)
+{
+    const auto &rawFlags = flags.rawConfig();
+    if (!(rawFlags.*aFlag))
+        return;
+    if (!(rawFlags.*otherFlag)) {
+        debugs(3, DBG_CRITICAL, "FATAL: " << PortOptionStr(aFlag) << " requires " << PortOptionStr(otherFlag) <<
+                " on " << PortKindStr(rawFlags) << ' ' << detail);
+        self_destruct();
+    }
+}
+
+bool
+AnyP::PortCfg::hasAll(const AnyP::TrafficModeFlags::List &list)
+{
+    const auto &rawFlags = flags.rawConfig();
+    for (const auto &p: list) {
+        if (!(rawFlags.*p))
+            return false;
+    }
+    return true;
+}
+
+void
+AnyP::PortCfg::checkFlags()
+{
+    using Flags = AnyP::TrafficModeFlags;
+    const auto &rawFlags = flags.rawConfig();
+
+    switch (rawFlags.portKind) {
+
+    case Flags::httpPort: {
+        if (rawFlags.accelSurrogate)
+            rejectFlags({&TrafficModeFlags::natIntercept, &TrafficModeFlags::tproxyIntercept}, "accel");
+        else
+            allowEither({&TrafficModeFlags::natIntercept, &TrafficModeFlags::tproxyIntercept});
```

Comment:
If possible, please simplify:

```suggestion
        allowEither({&TrafficModeFlags::natIntercept, &TrafficModeFlags::tproxyIntercept, &TrafficModeFlags::accelSurrogate});
```